File: lib/src/main/java/com/sha/formvalidator/validator/NumericRangeValidator.java
package com.sha.formvalidator.validator;

import android.widget.EditText;

/**
 * A validator that returns true only if the input field contains only numbers
 * and the number is within the given range.
 *
 */
public class NumericRangeValidator extends Validator {

    private long min, max;

    public NumericRangeValidator(String errorMessage, long min, long max) {
        super(errorMessage);
        this.min = min;
        this.max = max;
    }

    public boolean isValid(EditText et) {
        try {
            long value = Long.parseLong(et.getText().toString());
            return value >= min && value <= max;
        } catch (NumberFormatException e) {
            return false;
        }
    }
}

File: lib/src/main/java/com/sha/formvalidator/textview/validator/NumericRangeValidator.kt
package com.sha.formvalidator.textview.validator

import android.widget.TextView

/**
 * A validator that returns true only if the input field contains only numbers
 * and the number is within the given range.
 *
 */
class NumericRangeValidator(errorMessage: String, private val min: Long, private val max: Long) : Validator(errorMessage) {

    override fun isValid(tv: TextView): Boolean {
        return try {
            val value = java.lang.Long.parseLong(tv.text.toString())
            value in min..max
        } catch (e: NumberFormatException) {
            false
        }

    }
}
